# <center> Netflix Data Cleaning Project

<img src="netflix.jpeg" width=2000 height=1000>

# <center> About Dataset
A cleaned and transformed version of the Netflix Titles dataset prepared for exploratory data analysis, dashboarding, and data science projects.

Included Files Main Dataset netflix_clean.csv Exploded Tables netflix_country_exploded.csv netflix_genre_exploded.csv Dashboard Summary Tables country_summary.csv country_type_summary.csv genre_summary.csv rating_summary.csv yearly_titles.csv movies_tv.csv top_cast_summary.csv kpi_summary.csv Data Cleaning and Transformations Missing value analysis Country explosion for multi-country records Genre explosion for multi-label genres Separation of movie duration and TV show seasons Consistent formatting and preprocessing Summary tables prepared for Power BI dashboards Possible Use Cases Exploratory Data Analysis (EDA) Dashboard creation Data visualization Machine learning projects Statistics practice Educational purposes Project Structure 01_Data_Understanding.ipynb 02_Data_Cleaning.ipynb 03_Exploratory_Data_Analysis.ipynb 04_Dashboard_Preparation.ipynb 05_Dashboard_Planning.ipynb

License

CC BY-SA 4.0

# <center> Introduction

This project focuses on cleaning the **Netflix Titles dataset** (`NetFlix-selected-columns.csv`), which contains metadata for 7,787 movies and TV shows available on Netflix. The dataset includes 10 attributes describing each title — such as `show_id`, `type`, `title`, `director`, `cast`, `country`, `date_added`, `release_year`, `rating`, and `duration` — offering a rich mix of categorical, textual, and numerical data suitable for a real-world data cleaning exercise.

Despite being a well-known public dataset, it is far from clean in its raw form. It suffers from common real-world data quality issues, including missing values in key columns (`director`, `cast`, `country`, `date_added`, and `rating`), inconsistent date formatting in `date_added`, an ambiguous `duration` field that represents different units depending on content type (minutes for movies vs. seasons for TV shows), and multi-valued fields (`cast`, `country`) packed into single cells.

The goal of this project is to systematically identify, assess, and resolve these inconsistencies — handling missing data appropriately, standardizing formats, and restructuring ambiguous or compound fields — in order to produce a clean, analysis-ready version of the dataset. This process reflects the essential first step of any real-world data analysis workflow, where raw data must be transformed into a reliable and consistent form before meaningful insights can be drawn.

In [5]:
import pandas as pd

In [6]:
import warnings
warnings.filterwarnings("ignore")

In [7]:
df = pd.read_csv("Netflix.csv")

In [5]:
df.head()

,show_id,type,title,director,cast,country,date_added,release_year,rating,duration
0,s1,TV Show,3%,NaN,"João Miguel, Bianca Comparato, Michel Gomes, R...",Brazil,14-Aug-20,2020,TV-MA,4
1,s10,Movie,1920,Vikram Bhatt,"Rajneesh Duggal, Adah Sharma, Indraneil Sengup...",India,15-Dec-17,2008,TV-MA,143
2,s100,Movie,3 Heroines,Iman Brotoseno,"Reza Rahadian, Bunga Citra Lestari, Tara Basro...",Indonesia,5-Jan-19,2016,TV-PG,124
3,s1000,Movie,Blue Mountain State: The Rise of Thadland,Lev L. Spiro,"Alan Ritchson, Darin Brooks, James Cade, Rob R...",United States,1-Mar-16,2016,R,90
4,s1001,TV Show,Blue Planet II,NaN,David Attenborough,United Kingdom,3-Dec-18,2017,TV-G,1


In [6]:
df.shape

(7787, 10)

In [7]:
df.size

77870

In [8]:
df.columns

Index(['show_id', 'type', 'title', 'director', 'cast', 'country', 'date_added',
       'release_year', 'rating', 'duration'],
      dtype='object')

In [9]:
df.dtypes

show_id         object
type            object
title           object
director        object
cast            object
country         object
date_added      object
release_year     int64
rating          object
duration         int64
dtype: object

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7787 entries, 0 to 7786
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       7787 non-null   object
 1   type          7787 non-null   object
 2   title         7787 non-null   object
 3   director      5398 non-null   object
 4   cast          7069 non-null   object
 5   country       7280 non-null   object
 6   date_added    7777 non-null   object
 7   release_year  7787 non-null   int64 
 8   rating        7780 non-null   object
 9   duration      7787 non-null   int64 
dtypes: int64(2), object(8)
memory usage: 608.5+ KB


In [11]:
df.describe()

,release_year,duration
count,7787.000000,7787.000000
mean,2013.932580,69.122769
std,8.757395,50.950743
min,1925.000000,1.000000
25%,2013.000000,2.000000
50%,2017.000000,88.000000
75%,2018.000000,106.000000
max,2021.000000,312.000000


In [12]:
df.isnull().sum()

show_id            0
type               0
title              0
director        2389
cast             718
country          507
date_added        10
release_year       0
rating             7
duration           0
dtype: int64

In [13]:
df.isnull().sum().sum()

np.int64(3631)

In [14]:
df["director"].fillna("unknown" , inplace = True)

In [15]:
df["cast"].fillna(df["cast"].mode()[0] , inplace = True)

In [16]:
df["country"].fillna(df["country"].mode()[0] , inplace = True)

In [17]:
df["date_added"].fillna(df["date_added"].mode()[0] , inplace = True)

In [18]:
df["rating"].fillna(df["rating"].mode()[0] , inplace = True)

In [19]:
df.isnull().sum()

show_id         0
type            0
title           0
director        0
cast            0
country         0
date_added      0
release_year    0
rating          0
duration        0
dtype: int64

In [20]:
df.duplicated().sum()

np.int64(0)

In [21]:
df.to_csv("New_Netflix.csv", index = False)

# <center> Conclusion

Before diving into cleaning, I explored the dataset using `shape`, `size`, `info()`, `describe()`, `head()`, and `tail()` to understand its structure, size, and data types. This exploration showed that the dataset is mostly categorical in nature, with `release_year` being the main numerical column worth statistical summary. The `info()` step also revealed missing values in six columns — `director`, `cast`, `country`, `date_added`, and `rating` — totaling 3,631 null entries. I then checked for duplicate rows using `duplicated().sum()`, which confirmed that the dataset had **no duplicates**, meaning every record is unique.

To handle the missing values, I filled `director` with "unknown" since a large portion of its values were missing and no reliable substitute existed, while `cast`, `country`, `date_added`, and `rating` were filled using their respective mode (most frequent value), as these columns had comparatively fewer missing entries. After this treatment, `isnull().sum()` confirmed that the dataset now has zero missing values across all columns.

Overall, the dataset started out unique but incomplete, and after handling the null values through appropriate imputation techniques, it is now both clean and duplicate-free — making it ready for further analysis or visualization.